README.md

```diff
@@ -128,6 +128,13 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
 Note :
 1. If you built your own hardware, change VendorID and/or ProductID in [CommunicationHandler.java](https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/CommunicationHandler.java)
 
+### Permissions Required
+
+1. Record_Audio : It is required because in oscilloscope we are accepting inputs from the mic so we need to add this permission in order to contiune the proper functioning of this feature.You can find its implementation in [AudioJack.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/AudioJack.java).
+2. Access_Fine_Location And Internet : They are required because in compass and Map we need to get the location access so that the full use of these features can be taken.You can find its implementation in [GPSLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/GPSLogger.java).
+3. Write_External_Storage : It is required for writing log files in the storage as the test results are stored in it for future analysis.
+4. Read_External_Storage : While writing logs in the storage,[CSVLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/CSVLogger.java) first checks whether there is any CSVLogger directory exist or not and that require this read permission and its another use comes while implementing LeakCanary as it require read storage permission to detect the memory leaks in application.  
```

Comment:
1. storage,[CSVLogger --> storage, [CSVLogger

Reply:
Point 3:
For storing log files from instruments that can be transferred out for future analysis.